Sources/Satori/ClientProtocol.swift

```diff
@@ -0,0 +1,128 @@
+
+	/// Get all experiments data.
+	///
+	/// - Parameters:
+	/// 	- session: The session of the user.
+	func getAllExperimentsAsync(session: Session, retryConfig: RetryConfiguration?) async throws -> Satori_Api_ExperimentList
+
+	/// Get specific experiments data.
+	///
+	/// - Parameters:
+	/// 	- session: The session of the user.
+	/// 	- names: Experiment names.
+	func getExperimentsAsync(session: Session, names: [String], retryConfig: RetryConfiguration?) async throws -> Satori_Api_ExperimentList
+
+	/// Get a single flag for this identity.
+	///
+	/// - Parameters:
+	/// 	- session: The session of the user.
+	/// 	- name: The name of the flag.
+	/// 	- defaultValue: The default value if the server is unreachable.
+	func getFlagAsync(session: Session, name: String, defaultValue: String?, retryConfig: RetryConfiguration?) async throws -> Satori_Api_Flag
+
+	/// List all available flags for this identity.
+	///
+	/// - Parameters:
+	/// 	- session: The session of the user.
+	/// 	- names: Flag names, if empty string all flags are returned.
+	func getFlagsAsync(session: Session, names: [String], retryConfig: RetryConfiguration?) async throws -> Satori_Api_FlagList
+
+	/// List available live events.
+	///
+	/// - Parameters:
+	/// 	- session: The session of the user.
+	/// 	- names: Live event names, if null or empty, all live events are returned.
+	func getLiveEventsAsync(session: Session, names: [String]?, retryConfig: RetryConfiguration?) async throws -> Satori_Api_LiveEventList
+
+	/// Identify a session with a new ID.
+	///
+	/// - Parameters:
+	/// 	- session: The session of the user.
+	/// 	- id: Identity ID to enrich the current session and return a new session. The old session will no longer be usable. Must be between eight and 128 characters (inclusive). Must be an alphanumeric string with only underscores and hyphens allowed.
+	/// 	- defaultProperties: The default properties.
+	/// 	- customProperties: The custom event properties.
+	func identifyAsync(session: Session, id: String, defaultProperties: [String: String], customProperties: [String: String], retryConfig: RetryConfiguration?) async throws -> Session
+
+	/// List properties associated with this identity.
+	///
+	/// - Parameters:
+	/// 	- session: The session of the user.
+	func listPropertiesAsync(session: Session, retryConfig: RetryConfiguration?) async throws -> Satori_Api_Properties
+
+	/// Update properties associated with this identity.
+	///
+	/// - Parameters:
+	/// 	- session: The session of the user.
+	/// 	- defaultProperties: The default properties to update.
+	/// 	- customProperties: The custom properties to update.
+	/// 	- recompute: Whether or not to recompute the user's audience membership immediately after property update.
+	func updatePropertiesAsync(session: Session, defaultProperties: [String: String], customProperties: [String: String], recompute: Bool?, retryConfig: RetryConfiguration?) async throws -> Void
```

Comment:
I think `Async` isn't as idiomatic in Swift method names as it is in C#? If we don't have it for the Nakama client methods in Swift then I think we can drop them in this client too

Reply:
💯